packages/react-native/src/Authenticator/hooks/useFieldValues/useFieldValues.ts

```diff
@@ -37,6 +52,22 @@ export default function useFieldValues<FieldType extends TypedField>({
     return getSanitizedTextFields(fields, componentName);
   }, [componentName, fields, isRadioFieldComponent]);
 
+  const runValidation = useCallback(
```

Comment:
While it is true that `runValidation` isn't passed as a prop and only used within the scope of this function (`useFieldValues`), since we are calling it on blur events, I thought it was worth memoizing.

Reply:
What is the benefit if it is not preventing rerenders?

Reply:
My thought is that by memoizing `runValidation`, we ensure the same function instance will be reused , since it's dependency array is empty. Do you see any downsides?

Reply:
More that it's unclear to me what benefits memoizing provides versus declaring `runValidation` outside the scope of `useFieldValues`? For example why not something like this:
```tsx
const runValidations = (field: TextFieldOptionsType, value: string | undefined): ValidationError => {
  const fieldErrors = [];
  if (field.required && !value) {
    fieldErrors.push(getRequiredFieldText());
  }
  if (field.type == 'email') {
    fieldErrors.push(isValidEmail(value) ? '' : getInvalidEmailText());
  }
  return { [field.name]: fieldErrors };
}

function useFieldValues(params) {
  // ...field stuff
  
  const onBlur: TextFieldOnBlur = (event) => {
    // call `onBlur` passed as text `field` option
    field.onBlur?.(event);
    
    // call machine blur handler
    handleBlur({ name, value: values[name] });
    
    setfieldValidationErrors(runValidation(field, values[name]));
  };
}
```

Reply:
I've played with this and it's rerendering the validation error on every blur when the error is the same. Will tinker a bit, think we need to only rerender when there are new errors [i.e. the validation errors have changed].

Reply:
Refactored, moved out of the scope of the hook.